src/Simplify.cpp

```diff
@@ -1444,22 +1496,22 @@ class Simplify : public IRMutator {
             mod_rem = modulus_remainder(ramp_a->base, alignment_info);
         }
 
-        if (is_zero(a) && !is_zero(b)) {
+        if (is_zero(b) && !op->type.is_float()) {
+            expr = indeterminate_expression_error(op->type);
+        } else if (is_zero(a)) {
             expr = a;
         } else if (is_one(b)) {
             expr = a;
-        } else if (equal(a, b) &&
-                   !is_zero(b)) {
+        } else if (equal(a, b)) {
             expr = make_one(op->type);
         } else if (const_int(a, &ia) &&
-                   const_int(b, &ib) && ib) {
+                   const_int(b, &ib)) {
             expr = IntImm::make(op->type, div_imp(ia, ib));
         } else if (const_uint(a, &ua) &&
-                   const_uint(b, &ub) && ub) {
+                   const_uint(b, &ub)) {
             expr = UIntImm::make(op->type, ua / ub);
         } else if (const_float(a, &fa) &&
-                   const_float(b, &fb) &&
-                   fb != 0.0f) {
```

Comment:
Above, this creates an indeterminate expression when the denominator is zero, only if the result is not a float. So, this is going to divide by 0 at compile time, where it used to not do that. Was this intentional? I'm not sure if the behavior will actually be different (dividing by 0 at compile time instead of runtime seems like it won't matter).